test/cppsuite/test_harness/workload/thread_context.h

```diff
@@ -124,24 +128,25 @@ class thread_context {
     /*
      * Generic update function, takes a collection_id and key, will generate the value.
      *
-     * Returns true if it successfully updates the key, false if it receives rollback from the API.
+     * Returns a boolean indicating if a rollback is required.
      */
     bool update(scoped_cursor &cursor, uint64_t collection_id, const std::string &key);
 
     /*
      * Generic insert function, takes a collection_id and key_id, will generate the value.
      *
-     * Returns true if it successfully inserts the key, false if it receives rollback from the API.
+     * Returns a boolean indicating if a rollback is required.
      */
     bool insert(scoped_cursor &cursor, uint64_t collection_id, uint64_t key_id);
 
     /*
      * Generic next function.
      *
-     * Handles rollback and not found internally, but will return the error code to the caller so
-     * the caller can distinguish between them.
+     * Handles not found internally by resetting the cursor.
```

Comment:
Instead of `not found`, can we explicitly say `Handles WT_NOTFOUND internally...` ?

Reply:
It does reset the cursor as well in the case of `WT_ROLLBACK`. Should we say then:
`Handles internal errors by resetting the cursor` ?

Reply:
It does not reset the cursor in that case.